app/Rules/WorkingLinks.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace App\Rules;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Pool;
+use GuzzleHttp\Psr7\Request;
+use Psr\Http\Message\ResponseInterface;
+
+class WorkingLinks extends Rule
```

Comment:
The rule name isn't very descriptive IMHO. I'd prefer something more verbose, for example, `NoBrokenImmediateLinks`, admittedly a big wordy.